packages/back-end/src/controllers/organizations.ts

```diff
@@ -577,7 +581,76 @@ export async function postNamespaces(
   await updateOrganization(org.id, {
     settings: {
       ...org.settings,
-      namespaces: [...namespaces, { name, description }],
+      namespaces: [...namespaces, { name, description, status }],
+    },
+  });
+
+  res.status(200).json({
+    status: 200,
+  });
+}
+
+export async function putNamespaces(
+  req: AuthRequest<{
+    name: string;
+    description: string;
+    status: "active" | "inactive";
+  }>,
+  res: Response
+) {
+  req.checkPermissions("organizationSettings");
+
+  const { name, description, status } = req.body;
+  const { org } = getOrgFromReq(req);
+
+  const namespaces = org.settings?.namespaces || [];
+
+  // Namespace with the same name already exists
+  if (namespaces.filter((n) => n.name === name).length === 0) {
+    throw new Error("Namespace not found.");
+  }
+  const updatedNamespaces = namespaces.map((n) => {
+    if (n.name === name) {
+      return { name, description, status };
+    }
+    return n;
+  });
+
+  await updateOrganization(org.id, {
+    settings: {
+      ...org.settings,
+      namespaces: updatedNamespaces,
+    },
+  });
+
```

Comment:
Add an audit log entry so we don't accidentally lose data if things become corrupted for any reason.

Reply:
done